esmvaltool/diag_scripts/perfmetrics/portrait_plot.py

```diff
@@ -0,0 +1,513 @@
+        halves = [
+            [(0.5, -0.5), (-0.5, -0.5), (-0.5, 0.5)],  # top left
+            [(0.5, -0.5), (0.5, 0.5), (-0.5, 0.5)],  # bottom right
+        ]
+        return halves[position]
+    quarters = [
+        [(-0.5, -0.5), (0, 0), (-0.5, 0.5)],  # left
+        [(0.5, -0.5), (0, 0), (0.5, 0.5)],  # right
+        [(-0.5, 0.5), (0, 0), (0.5, 0.5)],  # bottom
+        [(-0.5, -0.5), (0, 0), (0.5, -0.5)],  # top
+    ]
+    return quarters[position]
+
+
+def plot_overlays(cfg, grid, data):
+    """Call overlay_reference for each split in data and each group in grid."""
+    split_count = data.shape[3]
+    group_count = data.shape[2]
+    for i in range(group_count):
+        if split_count < 2:
+            log.debug("No additional splits for overlay.")
+            break
+        if split_count > 4:
+            log.warning("Too many splits for overlay, only 3 will be plotted.")
+        group_data = data.isel({cfg["group_by"]: i})
+        group_data = group_data.dropna(cfg["x_by"], how="all")
+        for sss in range(split_count):
+            split = group_data.isel({cfg["split_by"]: sss})
+            split_label = split.coords[cfg["split_by"]].values.item()
+            if split_label == cfg["default_split"]:
+                log.debug("Skipping default split for overlay.")
+                continue
+            nodes = get_triangle_nodes(sss, split_count)
+            overlay_reference(cfg, grid[i], split.values.T, nodes)
+
+
+def plot(cfg, data):
+    """Create figure with subplots for each group.
+
+    sets same color range and  overlays additional references based on
+    the content of data (xr.DataArray)
+    """
+    fig = plt.figure(1, cfg.get("figsize", (5.5, 3.5)))
+    group_count = len(data.coords[cfg["group_by"]])
+    grid = ImageGrid(
+        fig,
+        111,  # similar to subplot(111)
+        cbar_mode="single",
+        cbar_location="right",
+        cbar_pad=0.1,
+        cbar_size=0.2,
+        nrows_ncols=(1, group_count),
+        axes_pad=0.1,
+    )
+    # remap colorbar to 10 discrete steps
+    cmap = mpl.cm.get_cmap(cfg.get("cmap", "RdYlBu_r"), 10)
+    cfg["plot_kwargs"]["cmap"] = cmap
+    for i in range(group_count):
+        group = data.isel({cfg["group_by"]: i})
+        group = group.dropna(cfg["x_by"], how="all")
+        title = None
+        if group_count > 1:
+            title = group.coords[cfg["group_by"]].values.item()
+        plot_group(cfg, grid[i], group, title=title)
+    # use same colorrange and colorbar for all subplots:
+    unify_limits(grid)
+    # set cb of first image as single cb for the figure
+    grid.cbar_axes[0].colorbar(grid[0].get_images()[0], **cfg["cbar_kwargs"])
+    if data.shape[3] > 1:
+        plot_overlays(cfg, grid, data)
+    if cfg["plot_legend"] and data.shape[3] > 1:
+        split_legend(cfg, grid, data)
+    basename = "portrait_plot"
+    fname = get_plot_filename(basename, cfg)
+    plt.savefig(fname, bbox_inches="tight", dpi=cfg["dpi"])
```

Comment:
[Provenance recording](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance) is missing here (the plot does not show up in the `index.html` file).